bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -584,16 +584,32 @@ impl RoomListItem {
     }
 
     /// Builds a `Room` FFI from an invited room without initializing its
-    /// internal timeline
+    /// internal timeline.
     ///
-    /// An error will be returned if the room is a state different than invited
+    /// An error will be returned if the room is a state different than invited.
     ///
     /// ⚠️ Holding on to this room instance after it has been joined is not
-    /// safe. Use `full_room` instead
+    /// safe. Use `full_room` instead.
+    #[deprecated(note = "Please use `room_without_timeline` instead.")]
     fn invited_room(&self) -> Result<Arc<Room>, RoomListError> {
-        if !matches!(self.membership(), Membership::Invited) {
+        self.room_without_timeline()
+    }
+
+    /// Builds a `Room` FFI from a room without initializing its internal
+    /// timeline.
+    ///
+    /// An error will be returned if the room is a state other than invited
+    /// or knocked, the 2 states that would match this use case.
+    ///
+    /// ⚠️ Holding on to this room instance after it has been joined is not
+    /// safe. Use `full_room` instead.
+    fn room_without_timeline(&self) -> Result<Arc<Room>, RoomListError> {
```

Comment:
Really not fond of this name, I see one of three ways to go from here (there can be more!):

1. Either rename this `preview_room()` or `stripped_room()`, as it's really a very basic stripped view of a room.
2. Rename `fn room()` to `fn room_with_timeline()`, and this one can be `fn room()` later.
3. (is kind of orthogonal, and could be in addition to (1) or (2)) It might be nice to return a `StrippedRoom` or `PreviewRoom` type instead of `Room`, from this function, to make it clear which functions are available and which are not. In fact, this might even get rid of the unused internal timeline instance.

What do you think?

Reply:
`preview_room` sounds fine to me. As for having a separate struct backing those, maybe we can create one, once we know all the values we need to expose through the FFI. I can try to get this working.